oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/directory/OakDirectory.java

```diff
@@ -341,7 +342,7 @@ private Set<String> getListing(){
         if (fileNames == null){
             fileNames = directoryBuilder.getChildNodeNames();
         }
-        Set<String> result = ImmutableSet.copyOf(fileNames);
+        Set<String> result = Collections.unmodifiableSet(SetUtils.toLinkedSet(fileNames));
```

Comment:
Also no need for immutability of null-safe lookups. This method is called only from the constructor, which just iterates over the contents of the set to add it to another set. I think this could be simplified. Directory listings may be long, so it may be worth to avoid copies here if possible.

Reply:
I would rather keep the optimizations separate from this ticket.
In this PR, I would try to keep the behavior as close as possible to the original.